### PR TITLE
Create nostr profile

### PR DESCRIPTION
**Pull request by benthecarman**
Allows you to create a nostr profile. Added a `sync_nostr` function that'll sync up their profile with primal, that way we can fetch changes made outside the app

### CODE REVIEW

File: mutiny-wasm/src/lib.rs [hunk 1]
```diff
@@ -1597,6 +1597,50 @@ impl MutinyWallet {
...
     }
 
+    /// Returns the user's nostr profile data
+    #[wasm_bindgen]
+    pub fn get_nostr_profile(&self) -> Result<JsValue, MutinyJsError> {
+        let profile = self.inner.nostr.get_profile()?;
+        Ok(JsValue::from_serde(&profile)?)
+    }
+
+    /// Returns the user's nostr profile data
```
Comment (futurepaul) [inline mutiny-wasm/src/lib.rs:1607]: dupe comment from above

**Comment by TonyGiorgio** [comment]
needs rebase

**Comment by futurepaul** [comment]
Can you rebase to master plz

**Comment by TonyGiorgio** [comment]
Can you manually cut a release without merging here? Should be able to from the command line, might as well make it 0.6.0-rc1

File: mutiny-core/src/labels.rs [hunk 2]
```diff
@@ -155,6 +155,20 @@ pub trait LabelStorage {
         }
         Ok(None)
     }
+    /// Finds a contact that has the given npub
+    fn get_contact_for_npub(
+        &self,
+        npub: XOnlyPublicKey,
+    ) -> Result<Option<(String, Contact)>, MutinyError> {
+        // todo this is not efficient, we should have a map of npub to contact
```
Comment (TonyGiorgio) [inline mutiny-core/src/labels.rs:163]: Does this todo need to happen here?

File: mutiny-core/src/lib.rs [hunk 3]
```diff
@@ -1637,6 +1639,42 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         Ok(())
     }
 
+    /// Uploads a profile pic to nostr.build and returns the uploaded file's URL
+    pub async fn upload_profile_pic(&self, image_bytes: Vec<u8>) -> Result<String, MutinyError> {
+        let client = reqwest::Client::new();
+
+        let form = Form::new().part("fileToUpload", Part::bytes(image_bytes));
+        let res: NostrBuildResult = client
+            .post("https://nostr.build/api/v2/upload/profile")
```
Comment (TonyGiorgio) [inline mutiny-core/src/lib.rs:1648]: Do we need to reach out to do this properly?

**Review by TonyGiorgio — APPROVED** [review]
Good to merge, just some questions if we want to handle them in follow ups or not.

File: mutiny-core/src/labels.rs (hunk 2 above)
Comment (benthecarman) [reply mutiny-core/src/labels.rs:163]: It's okay for now, not terribly inefficient, but we should fix before going to prod with this

File: mutiny-core/src/lib.rs (hunk 3 above)
Comment (benthecarman) [reply mutiny-core/src/lib.rs:1648]: Talked to their team already and this is what they recommened that we do

**Pull request merged**